### DISCUSSION TITLE
USB -C not working on version 2.1.3

### DISCUSSION THREAD

**Discussion by kickfliph** [mriscoc/Ender3V2S1#1234] [Q&A]
Hello there,

I'm new to the 3d printer world. I got an Ender 3S1 a few weeks ago second-hand almost new from a good friend of Mime and it came with this firmware pre-install. It has been working/printing very well so far until now I am looking to get Octoprint to work with the printer and my Raspberry Zero W2, I bought a new USB-C to USB-mini cable and I did the installation of Octoprint using this cable so I don't believe it is the problem but, once I get to connect the cable to the printer nothing is happening the RI even turs on. So my question is: Is his can be related to the firmware limitation or is something wrong with my printer?

I really appreciate all your help, if there is anything else needed please let me know.

Regards,

**Comment by mriscoc** [comment]
Check: https://github.com/mriscoc/Ender3V2S1/wiki/Octoprint

**Reply by kickfliph** [reply]
Hello there,

Thanks for replying. So I need to source the power of the Rasberry some please else because this USB-C in front will not provide the voltage to turn it on, Is that right? 

Regards,

**Reply by mriscoc** [reply]
Block the +VUSB from the USB cable, that is dangerous for your printer board.

**Comment by Crazy-Charles** [comment]
Just as a heads-up the Ender 3 S1 V24S1-301 board has no USB 5V connection to the board.

**Reply by kickfliph** [reply]
Hello there,

Thanks for your reply, I appreciate it. I'm not sure if that's the model of my printer because I don't see that anywhere but, this is what the software displays "Ender 3S1-F4-UBL", maybe this can help to confirm if what you are saying applies to me.

Thanks again.

**Reply by Crazy-Charles** [reply]
Yes it will apply to your printer, check on the Creality website to see if it looks the same, the firmware will have been compiled with a USB port speed of 250000.

**Reply by kickfliph** [reply]
Hey, thanks again,

So my goal is to have Octoprint running on the printer Now, based on what you just told me I need to; 

1.- Open the bottom of printer and compare the board with the one on the website.
2.- Download the latest version of his software and compile the USB speed port to 250000 
3.- Due fact that the USB port of this board does not provide 5V to power my Raspberry PI I'll have to provide power place else and connect it to the printer
4.- Run Octoprint.

Question: Even so I still need to remove the 5V pin from the USB cable to prevent my board to get damage?  

I 'm newbie, If I'm missing anything please let me know I would like to have this working but minimizing all risks possible.

Regards,

**Reply by Crazy-Charles** [reply]
1 - yes and also compare the overall appearance of the printer.
2 - 250000 is the default speed, you can either compile your own using mriscoc's tools in the https://github.com/mriscoc/Special_Configurations repository or download a precompiled version https://github.com/mriscoc/Ender3V2S1/releases/tag/20240122
3 - yes, the Pi will need its own power supply
4- Or try a terminal program first, i use Putty for serial port testing and once confirmed working, then try Octoprint.

Answer - no you don't, the motherboard in the printer will not allow 5V in or out, the pins are isolated, so no need to modify the cable.

Good luck!

**Reply by kickfliph** [reply]
It's working now, Thanks! I really appreciate it.
